### PR TITLE
Add New KBs

### LINKED ISSUE #1296: [KB] Add GitHub token permissions for EndBug/add-and-commit Action
Knowledge Base is missing for EndBug/add-and-commit.

### LINKED ISSUE #1307: [KB] Add GitHub token permissions for svenstaro/upload-release-action Action
Knowledge Base is missing for svenstaro/upload-release-action.

### LINKED ISSUE #1309: [KB] Add GitHub token permissions for benc-uk/workflow-dispatch Action
Knowledge Base is missing for benc-uk/workflow-dispatch.

### PR DESCRIPTION
**Pull request by vandana41**
Fixes #1296 Fixes #1307 Fixes #1309

### CODE REVIEW

**Comment by vandana41** [comment]
@h0x0er

File: knowledge-base/actions/svenstaro/action-security.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+name: 'Upload files to a GitHub release'
```
Comment (h0x0er) [inline knowledge-base/actions/svenstaro/action-security.yml:1]: @vandana41 please correct the `file-path`, the correct one would be `knowledge-base/actions/svenstaro/upload-release-action/action-security.yml`

File: knowledge-base/actions/endbug/add-and-commit/action-security.yml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+name: Add & Commit
+#Note: github token not used
```
Comment (h0x0er) [inline knowledge-base/actions/endbug/add-and-commit/action-security.yml:2]: @vandana41 this uses `github-token` [check_here](https://github.com/EndBug/add-and-commit/blob/main/action.yml#L64), please do the analysis

File: knowledge-base/actions/benc-uk/action-security.yml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+name: 'Workflow Dispatch'
```
Comment (h0x0er) [inline knowledge-base/actions/benc-uk/action-security.yml:1]: @vandana41 please correct the `filePath` to `knowledge-base/actions/benc-uk/workflow-dispatch/action-security.yml`

File: knowledge-base/actions/benc-uk/action-security.yml [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+github-token:
+  action-input:
+    input: token
+    is-default: false
+  permissions:
+    actions: read
```
Comment (h0x0er) [inline knowledge-base/actions/benc-uk/action-security.yml:7]: @vandana41  if you [check_this](https://github.com/benc-uk/workflow-dispatch/blob/827565b908f387ffd483c84312273ae185c06c8a/src/main.ts#L51), a post request is being sent to dispatch a workflow. As `dispatch_workflow` comes under `actions` [check_here](https://docs.github.com/en/rest/actions/workflows#create-a-workflow-dispatch-event), so the permission would be `actions: write`. And the reason would be `actions-reason: to list and dispatch workflows`

**Pull request closed without merging**